src/resources/views/crud/fields/select2_from_ajax.blade.php

```diff
@@ -58,115 +57,152 @@
-    <!-- include select2 js-->
-    <script src="{{ asset('packages/select2/dist/js/select2.full.min.js') }}"></script>
-    @if (app()->getLocale() !== 'en')
-    <script src="{{ asset('packages/select2/dist/js/i18n/' . app()->getLocale() . '.js') }}"></script>
-    @endif
-    @endpush
-
-@endif
-
-<!-- include field specific select2 js-->
-@push('crud_fields_scripts')
-<script>
-    function bpFieldInitSelect2FromAjaxElement(element) {
-        var form = element.closest('form');
-        var $placeholder = element.attr('data-placeholder');
-        var $minimumInputLength = element.attr('data-minimum-input-length');
-        var $dataSource = element.attr('data-data-source');
-        var $method = element.attr('data-method');
-        var $fieldAttribute = element.attr('data-field-attribute');
-        var $connectedEntityKeyName = element.attr('data-connected-entity-key-name');
-        var $includeAllFormFields = element.attr('data-include-all-form-fields')=='false' ? false : true;
-        var $allowClear = element.attr('data-column-nullable') == 'true' ? true : false;
-        var $dependencies = JSON.parse(element.attr('data-dependencies'));
-        var $ajaxDelay = element.attr('data-ajax-delay');
-
-        if (!$(element).hasClass("select2-hidden-accessible"))
-        {
-            $(element).select2({
-                theme: 'bootstrap',
-                multiple: false,
-                placeholder: $placeholder,
-                minimumInputLength: $minimumInputLength,
-                allowClear: $allowClear,
-                ajax: {
-                    url: $dataSource,
-                    type: $method,
-                    dataType: 'json',
-                    delay: $ajaxDelay,
-                    data: function (params) {
-                        if ($includeAllFormFields) {
-                            return {
-                                q: params.term, // search term
-                                page: params.page, // pagination
-                                form: form.serializeArray() // all other form inputs
-                            };
-                        } else {
-                            return {
-                                q: params.term, // search term
-                                page: params.page, // pagination
-                            };
-                        }
-                    },
-                    processResults: function (data, params) {
-                        params.page = params.page || 1;
-
-                        var result = {
-                            results: $.map(data.data, function (item) {
-                                textField = $fieldAttribute;
-                                return {
-                                    text: item[textField],
-                                    id: item[$connectedEntityKeyName]
+        <!-- select2_from_ajax field type js -->
+        @loadJsOnce('packages/select2/dist/js/select2.full.min.js')
```

Comment:
I'v fixed #2652 in the hope we can merge it also. 

It will fix issues in some other places like select_and_order and having the order functionality working on repeatable elements.

Reply:
I saw that you submit that, true. Haven't given it a proper look yet, and I don't plan to do so until we launch 4.1. I'm pretty sure if we choose to include that too, we'll end up delaying the launch by 1-2 more weeks and I really don't want to do that.

Best to admit we've crammed enough features into 4.1 already, and just ship it 😄 ⛵